src/csharp/Grpc.Core.Tests/Interceptors/ServerInterceptorTest.cs

```diff
@@ -0,0 +1,78 @@
+#region Copyright notice and license
+
+// Copyright 2017 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using Grpc.Core.Internal;
+using Grpc.Core.Tests;
+using Grpc.Core.Utils;
+using NUnit.Framework;
+
+namespace Grpc.Core.Interceptors.Tests
+{
+    public class ServerInterceptorTest
+    {
+        const string Host = "127.0.0.1";
+
+        private class AddRequestHeaderServerInterceptor : ServerInterceptor
+        {
+            readonly Metadata.Entry header;
+
+            public AddRequestHeaderServerInterceptor(string key, string value)
+            {
+                this.header = new Metadata.Entry(key, value);
+            }
+            public override async Task<TResponse> UnaryServerHandler<TRequest, TResponse>(TRequest request, ServerCallContext context, UnaryServerMethod<TRequest, TResponse> next)
```

Comment:
This method is async and `return await next()`, while the client test is non-async-but-still-Task and just returns `next()`. Is the difference intentional?

Reply:
Which client method? Line 69?

Reply:
I was referring to difference between the client interceprot test 
 (in [`ClientInterceptorTest.AddHeaderClientInterceptor.BlockingUnaryCall`](https://github.com/grpc/grpc/pull/12613/files#diff-000199ab016d959fb86fd61611223f6eR42)) and the server interceptor test.

The relevant client test code is:
```
public override TResponse BlockingUnaryCall<TRequest, TResponse>(Method<TRequest, TResponse> method, string host, CallOptions options, TRequest request, Func<Method<TRequest, TResponse>, string, CallOptions, TRequest, TResponse> next)
{
    options.Headers.Add(this.header);
    return next(method, host, options, request);
}
```

It just returns a Task, while in the similar `ServerInterceptorTest.AddRequestHeaderServerInterceptor.UnaryServerHandler`, the method is marked `async` and has the pattern `return await next(...)`. Since there's no other async action in this method, I would have expected a non-async method that just `return next(...);`

Is there something special about the server-side or is this an unintentional difference?

Reply:
Thanks for pointing it out. Fixed.